Mathlib/CategoryTheory/ChosenFiniteProducts.lean

```diff
@@ -768,8 +803,7 @@ variable {C : Type u} [Category.{v} C] [ChosenFiniteProducts C]
   [Limits.PreservesFiniteProducts F] [Limits.PreservesFiniteProducts G]
 
 attribute [local instance] Functor.monoidalOfChosenFiniteProducts in
-theorem monoidal_of_preservesFiniteProducts (α : F ⟶ G) :
-    NatTrans.IsMonoidal α where
+instance monoidal_of_preservesFiniteProducts (α : F ⟶ G) : NatTrans.IsMonoidal α where
```

Comment:
This one probably shouldn't be a global instance given that it uses a local instance for data. The design here, I'm pretty sure, is that the Chosen => Monoidal path should never be available by default for instance search.

Reply:
I understand *why* it was not marked as an instance, but I disagree that there's any issue here. `NatTrans.monoidal_of_preservesFiniteProducts` is an instance that talks about the specific instance `Functor.monoidalOfChosenFiniteProducts`. It won't fire in the absence of that instance